src/back/wgsl/writer.rs

```diff
@@ -82,6 +84,13 @@ impl<W: Write> Writer<W> {
     pub fn write(&mut self, module: &Module, info: &ModuleInfo) -> BackendResult {
         self.reset(module);
 
+        // Save all ep result types
+        for (_, ep) in module.entry_points.iter().enumerate() {
+            if let Some(ref result) = ep.function.result {
+                self.ep_results.insert(result.ty, ep.stage);
```

Comment:
what if two entry points of different stages happen to have the same types?

Reply:
Whoops. Thank you!